### PR TITLE
Add expandable @actions endpoint.

### PR DESCRIPTION
**Pull request by csenger**

### CODE REVIEW

File: docs/source/actions.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Portal Actions
+================
+
+You can get the available actions for a user on a context with the
```
Comment (tisto) [inline docs/source/actions.rst]: @csenger we usually have one or two sentences for devs who are not familiar with Plone telling them what this feature is about. We might also want to link to https://docs.plone.org/develop/plone/functionality/actions.html.

File: src/plone/restapi/services/actions/get.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        if self.request.get('translate'):
+            return translate(msg, context=self.request)
+        return msg
+
+    def reply(self):
+        context_state = api.content.get_view(name='plone_context_state',
+                                             context=self.context,
+                                             request=self.request)
+        data = {}
+        for category in ['object', 'object_buttons', 'user']:
```
Comment (tisto) [inline src/plone/restapi/services/actions/get.py]: @sneridagh @robgietema do we want to restrict the API to just expose those categories?

File: docs/source/actions.rst (hunk 1 above)
Comment (csenger) [reply docs/source/actions.rst]: I extended the documentation.

File: src/plone/restapi/services/actions/get.py (hunk 2 above)
Comment (csenger) [reply src/plone/restapi/services/actions/get.py]: The implementation now returns all category and can be limited to certain categories with parameter.

**Comment by sneridagh** [comment]
@tisto we need this in order to approve https://github.com/plone/plone-react/pull/118

File: src/plone/restapi/services/actions/get.py [hunk 3]
```diff
@@ -34,4 +47,20 @@ def reply(self):
                     'icon': action['icon'],
                 })
             data[category] = category_action_data
-        return data
+        return {'actions': data}
```
Comment (tisto) [inline src/plone/restapi/services/actions/get.py]: @csenger @sneridagh I had to wrap the data into actions to make it work with the expansion mechanism. I would have expected a test failure in test_actions but that did not happen at all. Need to investigate this further...

Comment (tisto) [reply src/plone/restapi/services/actions/get.py]: @csenger @sneridagh forget about what I wrote. This is specific to the expansion mechanism. I got confused. I think this is an indicator that we might want to re-think the entire expansion API design at some point...Anyways, I think except the docs we are good now...

**Comment by tisto** [comment]
@csenger @sneridagh except for the docs we are good now. Can you please do a quick last review? Then I can merge and release. I will take care of the docs when I watch the Barca game tonight (don't spoil me @sneridagh ;))

File: docs/source/actions.rst [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+Each actions defines an id, a title, the required
+permissions and a condition that will be checked to decide if the action
+will be available for a user.
+Actions are sorted by categories.
+
+Actions can be used to build UI elements that adapt to the available actions.
+An example is the Plone toolbar where the "object_tabs" (view, edit, folder contents, sharing)
+and the "user_actions" (login, logout, preferences) are used to display the user only the actions that are allowed for the currently logged in user.
+
+The available actions for the currently logge in user can be retrieved
```
Comment (sneridagh) [inline docs/source/actions.rst]: "logged"

File: src/plone/restapi/services/actions/get.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    def __call__(self, expand=False):
+        result = {
+            'actions': {
+                '@id': '{}/@actions'.format(self.context.absolute_url()),
+            },
+        }
+        if not expand:
+            return result
+
+        context_state = api.content.get_view(
```
Comment (sneridagh) [inline src/plone/restapi/services/actions/get.py]: Are we finally allowing plone.api in the code?

Comment (tisto) [reply src/plone/restapi/services/actions/get.py]: No. @csenger could you fix that please?

File: docs/source/actions.rst (hunk 4 above)
Comment (tisto) [reply docs/source/actions.rst]: Fixed.

**Pull request merged**